### ISSUE TITLE
Binaryen failures on Windows

### ISSUE THREAD

**Issue by juj** [emscripten-core/emscripten#4211]
After https://git.io/vV33C and https://github.com/WebAssembly/binaryen/pull/296, getting further on `asm2.test_binaryen` on Windows, however, now I'm getting into some bit mismatch:

```
C:\code\emsdk\emscripten\incoming>python tests/runner.py asm2.test_binaryen
DEBUG:root:failed to use vanilla file, will re-check: too many values to unpack
DEBUG:root:testing for asm.js target, because if not present (i.e. this is plain vanilla llvm, not emscripten fastcomp), we will use the wasm target instead (set EMCC_WASM_BACKEND to skip this check)
DEBUG:root:check tells us to use asm.js backend
WARNING:root:use EM_ALL_ENGINES=1 in the env to run against all JS engines, which is slower but provides more coverage
INFO:root:Blocking on process 13020: ['C:/code/emsdk/node/4.1.1_64bit/bin/node', 'C:\\code\\emsdk\\emscripten\\incoming\\src\\hello_world.js'] until it finishes.
INFO:root:Process 13020 finished after 0.138999938965 seconds. Exit code: 0
test_binaryen (test_core.asm2) ... (checking sanity from test runner)
INFO:root:(Emscripten: Running sanity checks)
INFO:root:Blocking on process 14364: ['C:/code/emsdk/node/4.1.1_64bit/bin/node', 'C:\\code\\emsdk\\emscripten\\incoming\\src\\hello_world.js'] until it finishes.
INFO:root:Process 14364 finished after 0.149000167847 seconds. Exit code: 0
DEBUG:root:failed to use vanilla file, will re-check: too many values to unpack
DEBUG:root:testing for asm.js target, because if not present (i.e. this is plain vanilla llvm, not emscripten fastcomp), we will use the wasm target instead (set EMCC_WASM_BACKEND to skip this check)
DEBUG:root:check tells us to use asm.js backend
WARNING:root:invocation: C:\code\emsdk\emscripten\incoming\emcc -Wno-dynamic-class-memaccess -Wno-format -Wno-format-extra-args -Wno-format-security -Wno-pointer-bool-conversion -Wno-unused-volatile-lvalue -Wno-c++11-compat-deprecated-writable-strings -Wno-invalid-pp-token -O2 -s GL_DEBUG=0 -s EXCEPTION_CATCHING_WHITELIST=[] -s INCLUDE_FULL_LIBRARY=0 -s BINARYEN=0 -s FINALIZE_ASM_JS=1 -s BINARYEN_ROOT="" -s SPLIT_MEMORY=0 -s RUNTIME_LINKED_LIBS=[] -s LIBRARY_DEBUG=0 -s ASSERTIONS=0 -s USE_SDL_IMAGE=1 -s SDL2_IMAGE_FORMATS=[] -s INLINING_LIMIT=0 -s MODULARIZE=0 -s NO_FILESYSTEM=0 -s ELIMINATE_DUPLICATE_FUNCTIONS=0 -s SOCKET_WEBRTC=0 -s USE_PTHREADS=0 -s USE_OGG=0 -s ABORTING_MALLOC=1 -s EXPORT_NAME="Module" -s AGGRESSIVE_VARIABLE_ELIMINATION=0 -s SIMPLIFY_IFS=1 -s ALIASING_FUNCTION_POINTERS=1 -s EMULATED_FUNCTION_POINTERS=0 -s EMSCRIPTEN_VERSION="" -s OPENAL_DEBUG=0 -s NO_EXIT_RUNTIME=0 -s RELOCATABLE=0 -s PRECISE_F32=0 -s DEBUG_TAGS_SHOWING=[] -s SAFE_HEAP_LOG=0 -s PROXY_TO_WORKER_FILENAME="" -s LIBRARY_DEPS_TO_AUTOEXPORT=["memcpy"] -s USE_CLOSURE_COMPILER=0 -s QUANTUM_SIZE=4 -s PRECISE_I64_MATH=1 -s SEPARATE_ASM=0 -s GLOBAL_BASE=-1 -s STACK_OVERFLOW_CHECK=0 -s FORCE_FILESYSTEM=0 -s LZ4=0 -s ASM_JS=1 -s WARN_UNALIGNED=0 -s EMULATE_FUNCTION_POINTER_CASTS=0 -s BINARYEN_SCRIPTS="" -s TOTAL_STACK=5242880 -s RESERVED_FUNCTION_POINTERS=0 -s EXCEPTION_DEBUG=0 -s FUNCTION_POINTER_ALIGNMENT=2 -s SHELL_FILE=0 -s GL_ASSERTIONS=0 -s BOOTSTRAPPING_STRUCT_INFO=0 -s MEM_INIT_METHOD=0 -s GL_TESTING=0 -s DETERMINISTIC=0 -s SAFE_HEAP=0 -s DEAD_FUNCTIONS=[] -s GL_FFP_ONLY=0 -s EMSCRIPTEN_TRACING=0 -s OPT_LEVEL=0 -s MEMFS_APPEND_TO_TYPED_ARRAYS=0 -s NO_DYNAMIC_EXECUTION=0 -s BUILD_AS_WORKER=0 -s EXPORT_ALL=0 -s EMTERPRETIFY_ADVISE=0 -s VERBOSE=0 -s DISABLE_EXCEPTION_CATCHING=1 -s PTHREADS_PROFILING=0 -s EMTERPRETIFY_WHITELIST=[] -s SKIP_STACK_IN_SMALL=1 -s USE_LIBPNG=0 -s EMTERPRETIFY_ASYNC=0 -s FULL_ES2=0 -s FULL_ES3=0 -s USE_WEBGL2=0 -s ELIMINATE_DUPLICATE_FUNCTIONS_DUMP_EQUIVALENT_FUNCTIONS=0 -s SIMD=0 -s PROXY_TO_WORKER=0 -s UNALIGNED_MEMORY=0 -s SWAPPABLE_ASM_MODULE=0 -s EXTRA_EXPORTED_RUNTIME_METHODS=[] -s ASYNCIFY=0 -s TOTAL_MEMORY=16777216 -s USE_SDL_NET=1 -s BINARYEN_IMPRECISE=0 -s SOCKET_DEBUG=0 -s ELIMINATE_DUPLICATE_FUNCTIONS_PASSES=5 -s EXPLICIT_ZEXT=0 -s FORCE_ALIGNED_MEMORY=0 -s HEADLESS=0 -s USE_SDL=1 -s USE_VORBIS=0 -s MAIN_MODULE=0 -s ASYNCIFY_WHITELIST=["qsort", "trinkle", "__toread", "__uflow", "__fwritex", "MUSL_vfprintf"] -s SAFE_SPLIT_MEMORY=0 -s WEBSOCKET_URL="ws:#" -s EMTERPRETIFY=0 -s DEBUG_LEVEL=0 -s SMALL_XHR_CHUNKS=0 -s USE_GLFW=2 -s WARN_ON_UNDEFINED_SYMBOLS=1 -s ORIGINAL_EXPORTED_FUNCTIONS=[] -s ONLY_MY_CODE=0 -s DEMANGLE_SUPPORT=0 -s RUNNING_JS_OPTS=0 -s WEBSOCKET_SUBPROTOCOL="binary" -s INVOKE_RUN=1 -s PTHREAD_HINT_NUM_CORES=4 -s PTHREAD_POOL_SIZE=0 -s DEFAULT_LIBRARY_FUNCS_TO_INCLUDE=["memcpy", "memset", "malloc", "free"] -s EXPORT_FUNCTION_TABLES=0 -s SIDE_MODULE=0 -s USE_SDL_TTF=1 -s GL_MAX_TEMP_BUFFER_SIZE=2097152 -s FS_LOG=0 -s STB_IMAGE=0 -s EXPORT_BINDINGS=0 -s SYSCALL_DEBUG=0 -s PGO=0 -s IN_TEST_HARNESS=0 -s ASYNCIFY_FUNCTIONS=["emscripten_sleep", "emscripten_wget", "emscripten_yield"] -s EMTERPRETIFY_BLACKLIST=[] -s EVAL_CTORS=0 -s USE_FREETYPE=0 -s DOUBLE_MODE=1 -s MAX_GLOBAL_ALIGN=-1 -s EMTERPRETIFY_FILE="" -s LEGACY_GL_EMULATION=0 -s RETAIN_COMPILER_SETTINGS=0 -s EXPORTED_FUNCTIONS=["_main"] -s EXPORTED_RUNTIME_METHODS=["FS_createFolder", "FS_createPath", "FS_createDataFile", "FS_createPreloadedFile", "FS_createLazyFile", "FS_createLink", "FS_createDevice", "FS_unlink", "Runtime", "ccall", "cwrap", "setValue", "getValue", "ALLOC_NORMAL", "ALLOC_STACK", "ALLOC_STATIC", "ALLOC_DYNAMIC", "ALLOC_NONE", "allocate", "getMemory", "Pointer_stringify", "AsciiToString", "stringToAscii", "UTF8ArrayToString", "UTF8ToString", "stringToUTF8Array", "stringToUTF8", "lengthBytesUTF8", "stackTrace", "addOnPreRun", "addOnInit", "addOnPreMain", "addOnExit", "addOnPostRun", "intArrayFromString", "intArrayToString", "writeStringToMemory", "writeArrayToMemory", "writeAsciiToMemory", "addRunDependency", "removeRunDependency"] -s ALLOW_MEMORY_GROWTH=0 -s USE_BULLET=0 -s CASE_INSENSITIVE_FS=0 -s USE_ZLIB=0 -s GL_UNSAFE_OPTS=1 -s BENCHMARK=0 -s BINARYEN_METHOD="" -s LINKABLE=0 -s WASM_BACKEND=0 -s BUILD_AS_SHARED_LIB=0 -s ERROR_ON_UNDEFINED_SYMBOLS=0 -s OUTLINING_LIMIT=0 -O2 -s BINARYEN=1 -s BINARYEN_METHOD="interpret-binary" -I c:/users/jukka/appdata/local/temp\emscripten_test_asm2_c0w7tm -I c:/users/jukka/appdata/local/temp\emscripten_test_asm2_c0w7tm\include c:/users/jukka/appdata/local/temp\emscripten_test_asm2_c0w7tm\src.cpp -o c:/users/jukka/appdata/local/temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js  (in c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm)
INFO:root:(Emscripten: Running sanity checks)
INFO:root:Blocking on process 9972: ['C:/code/emsdk/node/4.1.1_64bit/bin/node', 'C:\\code\\emsdk\\emscripten\\incoming\\src\\hello_world.js'] until it finishes.
INFO:root:Process 9972 finished after 0.151000022888 seconds. Exit code: 0
DEBUG:root:compiling to bitcode
DEBUG:root:emcc step "parse arguments and setup" took 0.02 seconds
DEBUG:root:compiling source file: c:/users/jukka/appdata/local/temp\emscripten_test_asm2_c0w7tm\src.cpp
DEBUG:root:including port: binaryen
DEBUG:root:    (at C:\Users\Jukka\.emscripten_ports\binaryen)
DEBUG:root:setting binaryen root to C:\Users\Jukka\.emscripten_ports\binaryen\binaryen-version_5
DEBUG:root:running: C:/code/emsdk/clang/fastcomp/build_incoming_vs2015_64/RelWithDebInfo/bin\clang++ -target asmjs-unknown-emscripten -D__EMSCRIPTEN_major__=1 -D__EMSCRIPTEN_minor__=36 -D__EMSCRIPTEN_tiny__=1 -Werror=implicit-function-declaration -nostdinc -Xclang -nobuiltininc -Xclang -nostdsysteminc -Xclang -isystemC:\code\emsdk\emscripten\incoming\system\local\include -Xclang -isystemC:\code\emsdk\emscripten\incoming\system\include\compat -Xclang -isystemC:\code\emsdk\emscripten\incoming\system\include -Xclang -isystemC:\code\emsdk\emscripten\incoming\system\include\emscripten -Xclang -isystemC:\code\emsdk\emscripten\incoming\system\include\libc -Xclang -isystemC:\code\emsdk\emscripten\incoming\system\lib\libc\musl\arch\emscripten -Xclang -isystemC:\code\emsdk\emscripten\incoming\system\include\libcxx -Xclang -isystemC:\code\emsdk\emscripten\incoming\system\lib\libcxxabi\include -Wno-dynamic-class-memaccess -Wno-format -Wno-format-extra-args -Wno-format-security -Wno-pointer-bool-conversion -Wno-unused-volatile-lvalue -Wno-c++11-compat-deprecated-writable-strings -Wno-invalid-pp-token -O2 -O2 -Ic:/users/jukka/appdata/local/temp\emscripten_test_asm2_c0w7tm -Ic:/users/jukka/appdata/local/temp\emscripten_test_asm2_c0w7tm\include -mllvm -disable-llvm-optzns -mllvm -disable-llvm-optzns -std=c++03 c:/users/jukka/appdata/local/temp\emscripten_test_asm2_c0w7tm\src.cpp -Xclang -isystemC:\code\emsdk\emscripten\incoming\system\include\SDL -emit-llvm -c -o c:/users/jukka/appdata/local/temp\tmplroucr\src_0.o
DEBUG:root:emcc step "bitcodeize inputs" took 0.15 seconds
DEBUG:root:optimizing c:/users/jukka/appdata/local/temp\emscripten_test_asm2_c0w7tm\src.cpp
DEBUG:root:emcc: LLVM opts: -O3 -disable-loop-vectorization -disable-slp-vectorization -vectorize-loops=false -vectorize-slp=false -vectorize-slp-aggressive=false  [num inputs: 1]
DEBUG:root:emcc step "process inputs" took 0.03 seconds
DEBUG:root:will generate JavaScript
DEBUG:root:including libc.bc
DEBUG:root:including dlmalloc.bc
DEBUG:root:including wasm-libc.bc
DEBUG:root:emcc step "calculate system libraries" took 0.04 seconds
DEBUG:root:linking: ['c:/users/jukka/appdata/local/temp\\tmplroucr\\src_0_1.o', 'C:\\Users\\Jukka\\.emscripten_cache\\asmjs\\libc.bc', 'C:\\Users\\Jukka\\.emscripten_cache\\asmjs\\dlmalloc.bc', 'C:\\Users\\Jukka\\.emscripten_cache\\asmjs\\wasm-libc.bc']
DEBUG:root:emcc step "link" took 0.01 seconds
DEBUG:root:saving intermediate processing steps to c:/users/jukka/appdata/local/temp\emscripten_temp
DEBUG:root:(not saving intermediate c:/users/jukka/appdata/local/temp\emscripten_temp\emcc-0-basebc.bc because deferring linking)
DEBUG:root:emcc: LLVM opts: -strip-debug -disable-verify -internalize -internalize-public-api-list=main,malloc,free,__errno_location,fflush -globaldce -disable-loop-vectorization -disable-slp-vectorization -vectorize-loops=false -vectorize-slp=false -vectorize-slp-aggressive=false  [num inputs: 4]
DEBUG:root:emcc step "post-link" took 0.20 seconds
DEBUG:root:LLVM => JS
INFO:root:Executing emscripten.py compiler with cmdline "c:/users/jukka/appdata/local/temp\tmplroucr\src.cpp.o.bc -o c:/users/jukka/appdata/local/temp\tmplroucr\src.cpp.o.bc.o.js -s GL_DEBUG=0 -s EXCEPTION_CATCHING_WHITELIST=[] -s INCLUDE_FULL_LIBRARY=0 -s BINARYEN=1 -s FINALIZE_ASM_JS=1 -s BINARYEN_ROOT="C:\\Users\\Jukka\\.emscripten_ports\\binaryen\\binaryen-version_5" -s SPLIT_MEMORY=0 -s RUNTIME_LINKED_LIBS=[] -s LIBRARY_DEBUG=0 -s ASSERTIONS=0 -s USE_SDL_IMAGE=1 -s SDL2_IMAGE_FORMATS=[] -s INLINING_LIMIT=0 -s MODULARIZE=0 -s NO_FILESYSTEM=0 -s ELIMINATE_DUPLICATE_FUNCTIONS=0 -s SOCKET_WEBRTC=0 -s USE_PTHREADS=0 -s USE_OGG=0 -s ABORTING_MALLOC=1 -s EXPORT_NAME="Module" -s AGGRESSIVE_VARIABLE_ELIMINATION=0 -s SIMPLIFY_IFS=1 -s ALIASING_FUNCTION_POINTERS=1 -s EMULATED_FUNCTION_POINTERS=0 -s EMSCRIPTEN_VERSION="1.36.1" -s OPENAL_DEBUG=0 -s NO_EXIT_RUNTIME=0 -s RELOCATABLE=0 -s PRECISE_F32=0 -s DEBUG_TAGS_SHOWING=[] -s SAFE_HEAP_LOG=0 -s PROXY_TO_WORKER_FILENAME="" -s LIBRARY_DEPS_TO_AUTOEXPORT=["memcpy"] -s USE_CLOSURE_COMPILER=0 -s QUANTUM_SIZE=4 -s PRECISE_I64_MATH=1 -s SEPARATE_ASM=0 -s GLOBAL_BASE=1024 -s STACK_OVERFLOW_CHECK=0 -s FORCE_FILESYSTEM=0 -s LZ4=0 -s ASM_JS=1 -s WARN_UNALIGNED=0 -s EMULATE_FUNCTION_POINTER_CASTS=0 -s BINARYEN_SCRIPTS="" -s TOTAL_STACK=5242880 -s RESERVED_FUNCTION_POINTERS=0 -s EXCEPTION_DEBUG=0 -s FUNCTION_POINTER_ALIGNMENT=2 -s SHELL_FILE=0 -s GL_ASSERTIONS=0 -s BOOTSTRAPPING_STRUCT_INFO=0 -s MEM_INIT_METHOD=1 -s GL_TESTING=0 -s DETERMINISTIC=0 -s SAFE_HEAP=0 -s DEAD_FUNCTIONS=[] -s GL_FFP_ONLY=0 -s EMSCRIPTEN_TRACING=0 -s OPT_LEVEL=2 -s MEMFS_APPEND_TO_TYPED_ARRAYS=0 -s NO_DYNAMIC_EXECUTION=0 -s BUILD_AS_WORKER=0 -s EXPORT_ALL=0 -s EMTERPRETIFY_ADVISE=0 -s VERBOSE=0 -s DISABLE_EXCEPTION_CATCHING=1 -s PTHREADS_PROFILING=0 -s EMTERPRETIFY_WHITELIST=[] -s SKIP_STACK_IN_SMALL=1 -s USE_LIBPNG=0 -s EMTERPRETIFY_ASYNC=0 -s FULL_ES2=0 -s FULL_ES3=0 -s USE_WEBGL2=0 -s ELIMINATE_DUPLICATE_FUNCTIONS_DUMP_EQUIVALENT_FUNCTIONS=0 -s SIMD=0 -s PROXY_TO_WORKER=0 -s UNALIGNED_MEMORY=0 -s SWAPPABLE_ASM_MODULE=0 -s EXTRA_EXPORTED_RUNTIME_METHODS=[] -s ASYNCIFY=0 -s TOTAL_MEMORY=16777216 -s USE_SDL_NET=1 -s BINARYEN_IMPRECISE=0 -s SOCKET_DEBUG=0 -s ELIMINATE_DUPLICATE_FUNCTIONS_PASSES=5 -s EXPLICIT_ZEXT=0 -s FORCE_ALIGNED_MEMORY=0 -s HEADLESS=0 -s USE_SDL=1 -s USE_VORBIS=0 -s MAIN_MODULE=0 -s ASYNCIFY_WHITELIST=["qsort", "trinkle", "__toread", "__uflow", "__fwritex", "MUSL_vfprintf"] -s SAFE_SPLIT_MEMORY=0 -s WEBSOCKET_URL="ws:#" -s EMTERPRETIFY=0 -s DEBUG_LEVEL=1 -s SMALL_XHR_CHUNKS=0 -s USE_GLFW=2 -s WARN_ON_UNDEFINED_SYMBOLS=1 -s ORIGINAL_EXPORTED_FUNCTIONS=["_main"] -s ONLY_MY_CODE=0 -s DEMANGLE_SUPPORT=0 -s RUNNING_JS_OPTS=1 -s WEBSOCKET_SUBPROTOCOL="binary" -s INVOKE_RUN=1 -s PTHREAD_HINT_NUM_CORES=4 -s PTHREAD_POOL_SIZE=0 -s DEFAULT_LIBRARY_FUNCS_TO_INCLUDE=["memcpy", "memset", "malloc", "free"] -s EXPORT_FUNCTION_TABLES=0 -s SIDE_MODULE=0 -s USE_SDL_TTF=1 -s GL_MAX_TEMP_BUFFER_SIZE=2097152 -s FS_LOG=0 -s STB_IMAGE=0 -s EXPORT_BINDINGS=0 -s SYSCALL_DEBUG=0 -s PGO=0 -s IN_TEST_HARNESS=0 -s ASYNCIFY_FUNCTIONS=["emscripten_sleep", "emscripten_wget", "emscripten_yield"] -s EMTERPRETIFY_BLACKLIST=[] -s EVAL_CTORS=0 -s USE_FREETYPE=0 -s DOUBLE_MODE=1 -s MAX_GLOBAL_ALIGN=-1 -s EMTERPRETIFY_FILE="" -s LEGACY_GL_EMULATION=0 -s RETAIN_COMPILER_SETTINGS=0 -s EXPORTED_FUNCTIONS=["_main", "_malloc", "_free", "___errno_location", "_fflush"] -s EXPORTED_RUNTIME_METHODS=["FS_createFolder", "FS_createPath", "FS_createDataFile", "FS_createPreloadedFile", "FS_createLazyFile", "FS_createLink", "FS_createDevice", "FS_unlink", "Runtime", "ccall", "cwrap", "setValue", "getValue", "ALLOC_NORMAL", "ALLOC_STACK", "ALLOC_STATIC", "ALLOC_DYNAMIC", "ALLOC_NONE", "allocate", "getMemory", "Pointer_stringify", "AsciiToString", "stringToAscii", "UTF8ArrayToString", "UTF8ToString", "stringToUTF8Array", "stringToUTF8", "lengthBytesUTF8", "stackTrace", "addOnPreRun", "addOnInit", "addOnPreMain", "addOnExit", "addOnPostRun", "intArrayFromString", "intArrayToString", "writeStringToMemory", "writeArrayToMemory", "writeAsciiToMemory", "addRunDependency", "removeRunDependency"] -s ALLOW_MEMORY_GROWTH=0 -s USE_BULLET=0 -s CASE_INSENSITIVE_FS=0 -s USE_ZLIB=0 -s GL_UNSAFE_OPTS=1 -s BENCHMARK=0 -s BINARYEN_METHOD="interpret-binary" -s LINKABLE=0 -s WASM_BACKEND=0 -s BUILD_AS_SHARED_LIB=0 -s ERROR_ON_UNDEFINED_SYMBOLS=0 -s OUTLINING_LIMIT=0 --libraries C:\code\emsdk\emscripten\incoming\src\library_pthread_stub.js"
DEBUG:root:emscript: llvm backend: C:/code/emsdk/clang/fastcomp/build_incoming_vs2015_64/RelWithDebInfo/bin\llc c:\users\jukka\appdata\local\temp\tmplroucr\src.cpp.o.bc -march=js -filetype=asm -o c:\users\jukka\appdata\local\temp\emscripten_temp\tmpoopfgb.4.js -emscripten-global-base=1024 -O2 -emscripten-wasm
INFO:root:Process 8640 finished after 0.0750000476837 seconds. Exit code: 0
DEBUG:root:  emscript: llvm backend took 0.0809998512268 seconds
DEBUG:root:emscript: js compiler glue
DEBUG:root:very limited syscalls (6, 146, 140, 54) so disabling full filesystem support
INFO:root:Blocking on process 17784: ['C:/code/emsdk/node/4.1.1_64bit/bin/node', 'C:\\code\\emsdk\\emscripten\\incoming\\src\\compiler.js', 'c:\\users\\jukka\\appdata\\local\\temp\\emscripten_temp\\tmpgqhtyl.txt', 'C:\\code\\emsdk\\emscripten\\incoming\\src\\library_pthread_stub.js'] until it finishes.
INFO:root:Process 17784 finished after 0.313000202179 seconds. Exit code: 0
DEBUG:root:  emscript: glue took 0.34500002861 seconds
DEBUG:root:emscript: python processing: function tables and exports
DEBUG:root:asm text sizes[[137068, 1715], 579, 130, 1134, 0, 402, 385, 460, 268, 550, 681]
DEBUG:root:  emscript: python processing: function tables and exports took 0.00800013542175 seconds
DEBUG:root:emscript: python processing: finalize
DEBUG:root:  emscript: python processing: finalize took 0.00300002098083 seconds
DEBUG:root:emcc step "emscript (llvm => executable code)" took 0.73 seconds
DEBUG:root:applying pre/postjses
DEBUG:root:wrote memory initialization to c:/users/jukka/appdata/local/temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js.mem
DEBUG:root:emcc step "source transforms" took 0.12 seconds
DEBUG:root:running js post-opts
DEBUG:root:applying js optimization passes: asm eliminate simplifyExpressions registerize minifyNames asmLastOpts last cleanup
chunkification: num funcs: 38 actual num chunks: 1 chunk size range: 140022 - 140022
DEBUG:root:config forcing native optimizer at C:/code/emsdk/emscripten/incoming_64bit_optimizer/RelWithDebInfo/optimizer.exe
DEBUG:root:js optimizer using native
DEBUG:root:config forcing native optimizer at C:/code/emsdk/emscripten/incoming_64bit_optimizer/RelWithDebInfo/optimizer.exe
run_on_chunk: ['C:/code/emsdk/emscripten/incoming_64bit_optimizer/RelWithDebInfo/optimizer.exe', 'c:\\users\\jukka\\appdata\\local\\temp\\emscripten_temp\\tmpzglgig.jsfunc_0.js', 'asm', 'eliminate', 'simplifyExpressions', 'registerize', 'minifyLocals', 'asmLastOpts', 'last']
running cleanup on shell code
DEBUG:root:emcc step "js opts" took 0.66 seconds
DEBUG:root:separating asm
DEBUG:root:failed to use vanilla file, will re-check: too many values to unpack
DEBUG:root:testing for asm.js target, because if not present (i.e. this is plain vanilla llvm, not emscripten fastcomp), we will use the wasm target instead (set EMCC_WASM_BACKEND to skip this check)
DEBUG:root:check tells us to use asm.js backend
DEBUG:root:integrating wasm.js polyfill interpreter
DEBUG:root:asm2wasm (asm.js => WebAssembly): C:\Users\Jukka\.emscripten_ports\binaryen\binaryen-version_5\bin\asm2wasm c:/users/jukka/appdata/local/temp\emscripten_test_asm2_c0w7tm\src.cpp.o.asm.js --mapped-globals=c:/users/jukka/appdata/local/temp\emscripten_test_asm2_c0w7tm\src.cpp.o.wast.mappedGlobals --total-memory=16777216
DEBUG:root:emcc step "asm2wasm" took 0.41 seconds
DEBUG:root:wasm-as (wasm => binary)
DEBUG:root:emcc step "final emitting" took 0.22 seconds
DEBUG:root:total time: 3.02 seconds
INFO:root:Blocking on process 11816: ['C:/code/emsdk/node/4.1.1_64bit/bin/node', 'c:/users/jukka/appdata/local/temp\\emscripten_test_asm2_c0w7tm\\src.cpp.o.js'] for 900 seconds
INFO:root:Process 11816 finished after 0.501000165939 seconds. Exit code: 7
(test did not pass in JS engine: ['C:/code/emsdk/node/4.1.1_64bit/bin/node'])
ERROR

======================================================================
ERROR: test_binaryen (test_core.asm2)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\code\emsdk\emscripten\incoming\tests\test_core.py", line 7988, in test_binaryen
    self.do_run(open(path_from_root('tests', 'hello_world.c')).read(), 'hello, world!')
  File "C:\code\emsdk\emscripten\incoming\tests\runner.py", line 612, in do_run
    raise e
Exception: Expected to find 'hello, world!' in '
c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:89717
   throw ex;
   ^
Assertion failed: x == y, at: src/wasm-binary.h,1289,verifyInt32 at Error
    at jsStackTrace (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:777:12)
    at stackTrace (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:791:11)
    at ___assert_fail (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:1386:207)
    at __ZN4wasm17WasmBinaryBuilder11verifyInt32Ei [wasm::WasmBinaryBuilder::verifyInt32(int)] (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:85864:82)
    at __ZN4wasm17WasmBinaryBuilder10readHeaderEv [wasm::WasmBinaryBuilder::readHeader()] (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:69486:2)
    at __ZN4wasm17WasmBinaryBuilder4readEv [wasm::WasmBinaryBuilder::read()] (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:46511:2)
    at Object._load_binary2wasm (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:57355:2)
    at Object.Module.asm (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:89616:32)
    at Object.<anonymous> (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:90906:1)
    at Module._compile (module.js:434:26)
', diff:

--- expected
+++ actual
@@ -1 +1,16 @@
-hello, world!
+
+c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:89717
+   throw ex;
+   ^
+Assertion failed: x == y, at: src/wasm-binary.h,1289,verifyInt32 at Error
+    at jsStackTrace (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:777:12)
+    at stackTrace (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:791:11)
+    at ___assert_fail (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:1386:207)
+    at __ZN4wasm17WasmBinaryBuilder11verifyInt32Ei [wasm::WasmBinaryBuilder::verifyInt32(int)] (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:85864:82)
+    at __ZN4wasm17WasmBinaryBuilder10readHeaderEv [wasm::WasmBinaryBuilder::readHeader()] (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:69486:2)
+    at __ZN4wasm17WasmBinaryBuilder4readEv [wasm::WasmBinaryBuilder::read()] (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:46511:2)
+    at Object._load_binary2wasm (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:57355:2)
+    at Object.Module.asm (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:89616:32)
+    at Object.<anonymous> (c:\users\jukka\appdata\local\temp\emscripten_test_asm2_c0w7tm\src.cpp.o.js:90906:1)
+    at Module._compile (module.js:434:26)
+



----------------------------------------------------------------------
Ran 1 test in 4.592s

FAILED (errors=1)
```

@kripken: do you have a thought where to start poking first? I looks like it's comparing the 4-byte magic value `0x6d736100` in `readHeader`, so at the very beginning. Perhaps this could be binary file write operations going wrong somewhere earlier?

**Comment by kripken** [comment]
Very strange. First thing, I would see if `check.py` passes in the binaryen repo.

**Comment by kripken** [comment]
My guess is that the file contains bad data, since we didn't write to it in binary mode (in wasm-as, which converts text to binary), which would be a windows-only issue. This PR should fix that: https://github.com/WebAssembly/binaryen/pull/303

**Comment by kripken** [comment]
Sadly no, now it fails to compile the fix ;)

````
  auto flags = std::ifstream::in;
  if (binary == Flags::Binary) flags |= std::ifstream::binary;


     3>C:\Users\clb\.emscripten_ports\binaryen\binaryen-version_7\src\support\file.cpp(27): error C2676: binary '|=' : 'std::_Iosb<int>::_Openmode' does not define this operator or a conversion to a type acceptable to the predefined operator [C:\Users\clb\.emscripten_ports\binaryen\binaryen-version_7\support.vcxproj]
````

wat?

**Comment by juj** [comment]
Reinstalled the Windows bot today on new hardware (yay), and looking at the latest binaryen build on the Windows bot, it's now saying

	DEBUG:root:asm2wasm (asm.js => WebAssembly): C:\Users\Jukka\.emscripten_ports\binaryen\binaryen-version_9\bin\asm2wasm c:/users/jukka/appdata/local/temp\emscripten_test_asm1_ve48qu\src.cpp.o.asm.js --mapped-globals=c:/users/jukka/appdata/local/temp\emscripten_test_asm1_ve48qu\src.cpp.o.wast.mappedGlobals --total-memory=16777216
	Fatal: Could not find pass: remove-unused-brs

	Traceback (most recent call last):
	  File "C:\code\emsdk\emscripten\incoming\emcc", line 13, in <module>
	    emcc.run()
	  File "C:\code\emsdk\emscripten\incoming\emcc.py", line 1924, in run
	    subprocess.check_call(cmd, stdout=open(wasm_text_target, 'w'))
	  File "C:\code\emsdk\python\2.7.5.3_64bit\lib\subprocess.py", line 542, in check_call
	    raise CalledProcessError(retcode, cmd)
	subprocess.CalledProcessError: Command '[u'C:\\Users\\Jukka\\.emscripten_ports\\binaryen\\binaryen-version_9\\bin\\asm2wasm', 'c:/users/jukka/appdata/local/temp\\emscripten_test_asm1_ve48qu\\src.cpp.o.asm.js', '--mapped-globals=c:/users/jukka/appdata/local/temp\\emscripten_test_asm1_ve48qu\\src.cpp.o.wast.mappedGlobals', '--total-memory=16777216']' returned non-zero exit status 1
	ERROR

I wonder if the binaryen version 9 is out of date what Emscripten incoming is expecting from trunk?

**Comment by kripken** [comment]
That looks like #577, a windows-specific linking issue. Basically, it can't find that `remove-unused-brs` pass because the linker either didn't link it in, or it was removed.

The passes register themselves using a global class instance, so it's a global ctor that keeps them from being dead code. Maybe we need something special for windows linking of such .o files? I know we use a special flag on linux.

(Meanwhile I updated the binaryen tag since it's been a while, but 99% sure it won't help here.)
